pkg/healthcheck/healthcheck.go

```diff
@@ -1265,6 +1265,21 @@ func (hc *HealthChecker) allCategories() []Category {
 		{
 			id: LinkerdDataPlaneChecks,
 			checkers: []Checker{
+				{
+					description: "data plane pods have service account volume mounted",
+					hintAnchor:  "",
+					warning:     true,
```

Comment:
@adleong, Do you think it makes more sense to replace warning with fatal? Because, eventually when we don't have a serviceaccount volume mount, the `l5d-data-plane-ready` is going to fail after this check. So I think we should just fail and stop the rest of the checks if this fails.